ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -0,0 +1,970 @@
+                self.LOGGER.logger("Connection successfully established!", self.module, 1, 'port_celery')
+            else:
+                self.LOGGER.logger("Connection FAILED to service Celery, please check 'RabbitMQ' and 'ovs-workers'?",
+                                   self.module, 0, 'port_celery')
+        else:
+            self.LOGGER.logger("RabbitMQ is not running/active on this server!", self.module, 5, 'port_celery')
+
+    def check_ovs_packages(self):
+        """
+        Checks the availability of packages for Open vStorage
+        """
+
+        self.LOGGER.logger("Checking OVS packages: ", self.module, 3, 'check_ovs_packages', False)
+
+        for package in self.openvstorageTotalPackageList:
+            result = self.utility.execute_bash_command("apt-cache policy %s" % package)
+            if len(result) != 1:
+                self.LOGGER.logger(
+                    "Package '%s' is present, with version '%s'" % (package, result[2].split(':')[1].strip()),
+                    self.module, 1, 'package_{0}'.format(package))
+            else:
+                self.LOGGER.logger("Package '{0}' is NOT present ...".format(package), self.module, 5,
+                                   'package_{0}'.format(package))
+        return None
+
+    def check_ovs_processes(self):
+        """
+        Checks the availability of processes for Open vStorage
+        """
+
+        self.LOGGER.logger("Checking LOCAL OVS services: ", self.module, 3, 'check_ovs_processes', False)
+
+        if self.service_manager:
+            for ovs_service in os.listdir("/etc/init"):
+                if ovs_service.startswith("ovs-"):
+                    process_name = ovs_service.split(".conf", 1)[0].strip()
+                    if self.utility.check_status_of_service(process_name):
+                        self.LOGGER.logger("Service '{0}' is running!".format(process_name), self.module, 1,
+                                           'process_{0}'.format(process_name))
+                    else:
+                        self.LOGGER.logger("Service '{0}' is NOT running, please check this... ".format(process_name),
+                                           self.module, 0, 'process_{0}'.format(process_name))
+        else:
+            self.LOGGER.logger("Other service managers than 'init' are not yet supported!", self.module, 4,
+                               'hc_process_supported', False)
+
+    def _method_handler(self, signum, frame):
+        """
+        Method handler for python signals, specifically used to time-out when a call to a process is taking too long.
+        Use in combination with: `signal.signal(signal.SIGALRM, self._method_handler)`
+
+        @raises Exception
+        """
+
+        warning = "SPOTTED a PROCESS who is taking to long! Signum: {0} ; Frame {1}".format(signum, frame)
+        self.LOGGER.logger(warning, self.module, 3, 'spotted_idle_process', False)
+        raise Exception(warning)
```

Comment:
Currently we've changed to a RunTimeError, this would make more sence than a broad Exception. 

Also this will be replaced in the near future by: `threading.Thread`